session.go

```diff
@@ -223,6 +223,27 @@ func (s *session) resend(msg *Message) bool {
 	return s.application.ToApp(msg, s.sessionID) == nil
 }
 
+// queueBatchAppsForSend will validate, persist, and queue the messages for send.
+func (s *session) queueBatchAppsForSend(msg []*Message) error {
+	s.sendMutex.Lock()
+	defer s.sendMutex.Unlock()
+
+	msgBytes, err := s.prepBatchAppMessagesForSend(msg)
+	if err != nil {
+		return err
+	}
+
+	for _, mb := range msgBytes {
+		s.toSend = append(s.toSend, mb)
+		select {
+		case s.messageEvent <- true:
```

Comment:
this can be moved out of the loop as we don't release the lock until we return so it is not very useful to notify each time